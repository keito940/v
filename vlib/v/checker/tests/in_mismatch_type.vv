type Int = int

type Str2 = string
type Str3 = Str2

fn main() {
	a_i := [1, 2, 3]
	a_s := ['1', '2', '3']
	m := {
		'test': 1
	}
	s := 'abcd'
	if 1 in a_s {
		println('ok')
	}
	if 2 in m {
		println('yeah')
	}
	if 3 in s {
		println('dope')
	}
	if `a` in s {
		println("oh no :'(")
	}
	if 1 in 12 {
		println('right')
	}
	if Int(2) in m {
		println('yeah')
	}
	if Str2('3') in a_i {
		println('sure')
	}
	if Str3('2') in a_i {
		println('all right')
	}
}
