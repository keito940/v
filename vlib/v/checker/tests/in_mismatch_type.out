vlib/v/checker/tests/in_mismatch_type.v:13:7: error: the data type on the left of `in` does not match the array item type
   11|     }
   12|     s := 'abcd'
   13|     if 1 in a_s {
                ~~
   14|         println('ok')
   15|     }
vlib/v/checker/tests/in_mismatch_type.v:16:7: error: the data type on the left of `in` does not match the map key type
   14|         println('ok')
   15|     }
   16|     if 2 in m {
                ~~
   17|         println('yeah')
   18|     }
vlib/v/checker/tests/in_mismatch_type.v:19:7: error: the data type on the left of `in` must be a string
   17|         println('yeah')
   18|     }
   19|     if 3 in s {
                ~~
   20|         println('dope')
   21|     }
vlib/v/checker/tests/in_mismatch_type.v:22:9: error: the data type on the left of `in` must be a string
   20|         println('dope')
   21|     }
   22|     if `a` in s {
                  ~~
   23|         println("oh no :'(")
   24|     }
vlib/v/checker/tests/in_mismatch_type.v:25:7: error: `in` can only be used with an array/map/string
   23|         println("oh no :'(")
   24|     }
   25|     if 1 in 12 {
                ~~
   26|         println('right')
   27|     }
vlib/v/checker/tests/in_mismatch_type.v:28:12: error: the data type on the left of `in` does not match the map key type
   26|         println('right')
   27|     }
   28|     if Int(2) in m {
                     ~~
   29|         println('yeah')
   30|     }
vlib/v/checker/tests/in_mismatch_type.v:31:15: error: the data type on the left of `in` does not match the array item type
   29|         println('yeah')
   30|     }
   31|     if Str2('3') in a_i {
                        ~~
   32|         println('sure')
   33|     }
vlib/v/checker/tests/in_mismatch_type.v:34:15: error: the data type on the left of `in` does not match the array item type
   32|         println('sure')
   33|     }
   34|     if Str3('2') in a_i {
                        ~~
   35|         println('all right')
   36|     }